x-pack/plugins/observability/server/lib/collectors/register.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import { fetcher } from './fetcher';
+import type { Usage } from './type';
+
+export function registerSloUsageCollector(usageCollection?: UsageCollectionSetup): void {
+  if (!usageCollection) {
+    return;
+  }
+
+  const sloUsageCollector = usageCollection.makeUsageCollector<Usage>({
+    type: 'slo',
+    schema: {
+      slo: {
+        total: {
+          type: 'long',
+          _meta: {
+            description: 'The total number of slos in the cluster',
+          },
+        },
+        by_status: {
+          enabled: {
+            type: 'long',
+            _meta: {
+              description: 'The number of enabled slos in the cluster',
+            },
+          },
+          disabled: {
+            type: 'long',
+            _meta: {
+              description: 'The number of disabled slos in the cluster',
+            },
+          },
+        },
+        by_sli_type: {
+          DYNAMIC_KEY: {
+            type: 'long',
+            _meta: {
+              description: 'The number of slos by sli type in the cluster',
+            },
+          },
+        },
+        by_rolling_duration: {
+          DYNAMIC_KEY: {
+            type: 'long',
+            _meta: {
+              description: 'The number of slos by rolling duration in the cluster',
```

Comment:
My only concern is the cardinality of these `DYNAMIC_KEY`s. We can post-process them during index time on the receiving side if necessary, though.

Reply:
Theoretically, it could be any duration (from 1d to 1y) but the UI only supports 7d, 30d and 90d. So we should be good for most cluster